### ISSUE TITLE
trainvae训练效果很差

### ISSUE THREAD

**Issue by s9rout** [tudelft/mavrl#8]
根据readme复现到trainvae过程中，无论是使用自己运行collect_data.py收集的数据还是使用项目中提供的lstm_dataset数据，训练vae时效果都不理想，loss波动严重，贴出部分输出如下
Train Epoch: 80 [0/400 (0%)]	Loss: 4442.062012
====> Epoch: 80 Average loss: 3920.0713
====> Test set loss: 4540.9381
Train Epoch: 81 [0/400 (0%)]	Loss: 6491.220215
====> Epoch: 81 Average loss: 6800.8358
====> Test set loss: 4545.3127
Train Epoch: 82 [0/400 (0%)]	Loss: 3707.228027
====> Epoch: 82 Average loss: 4176.3337
====> Test set loss: 4565.2113
Train Epoch: 83 [0/400 (0%)]	Loss: 3795.062744
====> Epoch: 83 Average loss: 4041.0923
====> Test set loss: 4569.2432
Train Epoch: 84 [0/400 (0%)]	Loss: 3782.567871
====> Epoch: 84 Average loss: 3472.0280
====> Test set loss: 4569.7635
Train Epoch: 85 [0/400 (0%)]	Loss: 5683.165039
====> Epoch: 85 Average loss: 5494.4209
====> Test set loss: 4598.1790
Train Epoch: 86 [0/400 (0%)]	Loss: 2001.957153
====> Epoch: 86 Average loss: 2553.6716
====> Test set loss: 4595.9354
Train Epoch: 87 [0/400 (0%)]	Loss: 4960.599609
====> Epoch: 87 Average loss: 4761.0453
====> Test set loss: 4579.0131
Train Epoch: 88 [0/400 (0%)]	Loss: 3414.880127
====> Epoch: 88 Average loss: 3207.4652
====> Test set loss: 4561.6961
Train Epoch: 89 [0/400 (0%)]	Loss: 4360.537598
====> Epoch: 89 Average loss: 4413.1440
====> Test set loss: 4569.0705
Train Epoch: 90 [0/400 (0%)]	Loss: 5056.951172
====> Epoch: 90 Average loss: 5029.2564
====> Test set loss: 4586.4703
Train Epoch: 91 [0/400 (0%)]	Loss: 3803.083984
====> Epoch: 91 Average loss: 3294.9556
====> Test set loss: 4584.8609
Train Epoch: 92 [0/400 (0%)]	Loss: 4470.384277
====> Epoch: 92 Average loss: 3997.2199
====> Test set loss: 4576.6551
Train Epoch: 93 [0/400 (0%)]	Loss: 3465.927734
====> Epoch: 93 Average loss: 3609.2566
====> Test set loss: 4585.1057
Train Epoch: 94 [0/400 (0%)]	Loss: 4982.423340
====> Epoch: 94 Average loss: 4368.8247
====> Test set loss: 4592.1392
Train Epoch: 95 [0/400 (0%)]	Loss: 2709.912598
====> Epoch: 95 Average loss: 2833.8211
====> Test set loss: 4590.0280
Train Epoch: 96 [0/400 (0%)]	Loss: 4688.292480
====> Epoch: 96 Average loss: 4587.2419
====> Test set loss: 4581.8330
Train Epoch: 97 [0/400 (0%)]	Loss: 3824.011475
====> Epoch: 97 Average loss: 4081.1715
====> Test set loss: 4580.6645
Train Epoch: 98 [0/400 (0%)]	Loss: 4193.021973
====> Epoch: 98 Average loss: 4130.1999
====> Test set loss: 4585.7657
Train Epoch: 99 [0/400 (0%)]	Loss: 5308.291992
====> Epoch: 99 Average loss: 5793.0313
====> Test set loss: 4575.7138
Train Epoch: 100 [0/400 (0%)]	Loss: 4783.895508
====> Epoch: 100 Average loss: 5051.3898
====> Test set loss: 4588.9565
End of Training because of early stopping at epoch 100

**Comment by NPU-yuhang** [comment]
Fluctuations in the loss curve are perfectly normal given the variance in data distribution across epochs. In fact, these oscillations often suggest that the model has reached convergence; you can plot the full loss curve to confirm this trend. It is also worth performing a visual check on the reconstructed images in `exp_vae/vae/samples` to see if the model can clearly distinguish between obstacles and blank areas. If the results are not yet optimal, fine-tuning the learning rate would be a good next step.

Here I also attach the loss curves in my side after 160 epoches training:

<img width="640" height="480" alt="Image" src="https://github.com/user-attachments/assets/fae52c2c-96af-4d65-b472-108748b4b156" />
